source/common/router/router.cc

```diff
@@ -100,19 +101,19 @@ void FilterUtility::setUpstreamScheme(Http::HeaderMap& headers, bool use_secure_
   }
 }
 
-bool FilterUtility::shouldShadow(const ShadowPolicy& policy, Runtime::Loader& runtime,
+bool FilterUtility::shouldShadow(const ShadowPolicyPtr& policy, Runtime::Loader& runtime,
                                  uint64_t stable_random) {
```

Comment:
nit: `ASSERT( policy != nullptr)`

Reply:
✅

Reply:
The style guide explicitly calls this out as a case where ASSERT shouldn't be used:

```
Use `ASSERT` liberally, but do not use it for things that will crash in an obvious way in a subsequent line.
E.g., do not do `ASSERT(foo != nullptr); foo->doSomething();`.
```

Instead I would probably have this function continue to take a ref to the policy itself and deref the pointer before calling `shouldShadow`.